manifests/config.pp

```diff
@@ -21,13 +21,24 @@
     "main_collective" => $main_collective
   }
 
+  # some config settings are set directly on the main class, they are
+  # things other classes need to function for example libdir is also
+  # used by the plugin installer.  Specifically these settings should
+  # never be overridden by settings from the hashes since any difference
+  # between server/client cfg and what ends up being used by the plugin
+  # installer will lead to plugins not being found.
+  $global_config = {
```

Comment:
yeah good catch, they just should not be in the hashes by default - the earlier code setting `$client_collectives` and `$server_collectives` already handles that - while letting people mess with them in the config hashes if they want.  These are only special because main collective is derived programatically by default I guess